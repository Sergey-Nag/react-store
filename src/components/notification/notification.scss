.notifications-wrapper .alert {
  counter-increment: alert;
  min-width: 200px;
  pointer-events: all;

  &::before {
    content: counter(alert);
    position: absolute;
    top: 0;
    left: 5px;
    opacity: 0.5;
  }
}