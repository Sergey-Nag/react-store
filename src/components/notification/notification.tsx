import { PropsWithChildren, useMemo } from "react";
import { ExclamationCircle, CheckCircle, InfoCircle } from 'react-bootstrap-icons';
import './notification.scss';
import cn from "classnames";
import { NotificationStyle } from "../../enums/notification-style";

export interface NotificationProps {
  type: NotificationStyle,
}

export function Notification({ type, children  }: PropsWithChildren<NotificationProps>) {
  const Icon = useMemo(() => {
    switch(type) {
      case NotificationStyle.error: return ExclamationCircle;
      case NotificationStyle.info: return InfoCircle;
      case NotificationStyle.default: return CheckCircle;
    }
  }, [type]);
  
  const alertClassName = cn('alert d-flex align-items-center', `alert-${type}`);

  return (
    <div className={alertClassName} role="alert">
      <Icon className="me-2" size="1.2rem" />
      <div className="text">
        {children}
      </div>
    </div>
  );
}
