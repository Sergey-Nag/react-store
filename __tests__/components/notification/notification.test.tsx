import { fireEvent, render, screen } from '@testing-library/react';
import { ReactElement } from 'react';
import { Notification, NotificationProps } from '../../../src/components/notification/notification';

const MockComponent = ({ children }: { children: ReactElement }) => 
  <div className='some-element'>{children}</div>;

describe('Notification', () => {
  beforeEach(() => {
    const root = document.createElement('div');
    root.id = 'root';
    document.body.append(root);
  });

  // it('Should render component as a direct child of the root component whatever the location of the component', () => {
  //   const { baseElement } = render(
  //     <MockComponent>
  //       <Notification>test</Notification>
  //     </MockComponent>
  //   );
    
  //   expect(baseElement).toMatchSnapshot();
  // });

  // describe('Should render different styles of notification depends on the alertStyle prop:', () => {
  //   it.each(['primary', 'secondary', 'info', 'danger'] as NotificationProps['alertStyle'][])('%s', (style) => {
  //     render(<Notification alertStyle={style}>test</Notification>);
      
  //     expect(screen.getByRole('alert')).toBeInTheDocument();
  //     expect(screen.getByRole('alert')).toHaveClass(`alert-${style}`);
  //   });
  // });

  // it('Should remove a specific alert component by clicking the close button', () => {
  //   render(<Notification>test</Notification>);

  //   expect(screen.getByRole('alert')).toBeInTheDocument();

  //   fireEvent.click(screen.getByLabelText('Close'));

  //   expect(screen.queryByRole('alert')).not.toBeInTheDocument();
  // }); 
});