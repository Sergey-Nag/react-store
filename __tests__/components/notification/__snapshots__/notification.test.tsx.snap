// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Should render the component with different types danger 1`] = `
<DocumentFragment>
  <div
    class="alert d-flex align-items-center alert-danger"
    role="alert"
  >
    <svg
      class="me-2"
      fill="currentColor"
      height="1.2rem"
      viewBox="0 0 16 16"
      width="1.2rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
      />
      <path
        d="M7.002 11a1 1 0 1 1 2 0 1 1 0 0 1-2 0zM7.1 4.995a.905.905 0 1 1 1.8 0l-.35 3.507a.552.552 0 0 1-1.1 0L7.1 4.995z"
      />
    </svg>
    <div
      class="text"
    >
      Test notifications
    </div>
  </div>
</DocumentFragment>
`;

exports[`Notification Should render the component with different types info 1`] = `
<DocumentFragment>
  <div
    class="alert d-flex align-items-center alert-info"
    role="alert"
  >
    <svg
      class="me-2"
      fill="currentColor"
      height="1.2rem"
      viewBox="0 0 16 16"
      width="1.2rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
      />
      <path
        d="m8.93 6.588-2.29.287-.082.38.45.083c.294.07.352.176.288.469l-.738 3.468c-.194.897.105 1.319.808 1.319.545 0 1.178-.252 1.465-.598l.088-.416c-.2.176-.492.246-.686.246-.275 0-.375-.193-.304-.533L8.93 6.588zM9 4.5a1 1 0 1 1-2 0 1 1 0 0 1 2 0z"
      />
    </svg>
    <div
      class="text"
    >
      Test notifications
    </div>
  </div>
</DocumentFragment>
`;

exports[`Notification Should render the component with different types primary 1`] = `
<DocumentFragment>
  <div
    class="alert d-flex align-items-center alert-primary"
    role="alert"
  >
    <svg
      class="me-2"
      fill="currentColor"
      height="1.2rem"
      viewBox="0 0 16 16"
      width="1.2rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
      />
      <path
        d="M10.97 4.97a.235.235 0 0 0-.02.022L7.477 9.417 5.384 7.323a.75.75 0 0 0-1.06 1.06L6.97 11.03a.75.75 0 0 0 1.079-.02l3.992-4.99a.75.75 0 0 0-1.071-1.05z"
      />
    </svg>
    <div
      class="text"
    >
      Test notifications
    </div>
  </div>
</DocumentFragment>
`;
